resolwe_bio/processes/variant_calling/snpeff.py

```diff
@@ -0,0 +1,236 @@
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/genialis/resolwebio/snpeff:2.1.1"},
+        },
+        "resources": {
+            "cores": 2,
+            "memory": 16384,
+        },
+    }
+    entity = {
+        "type": "sample",
+    }
+    category = "Other"
+    data_name = "Annotated variants (SnpEff)"
+    version = "1.0.0"
+    scheduling_class = SchedulingClass.BATCH
+    persistence = Persistence.CACHED
+
+    class Input:
+        """Input fields to SnpEff process."""
+
+        variants = DataField(
+            data_type="variants:vcf",
+            label="Variants (VCF)",
+        )
+        database = StringField(
+            label="snpEff database",
+            default="GRCh37.75",
+            choices=[("GRCh37.75", "GRCh37.75"), ("GRCh38.99", "GRCh38.99")],
+        )
+        dbsnp = DataField(
+            data_type="variants:vcf",
+            label="Known variants",
+            description="List of known variants for annotation.",
+            required=False,
+        )
+        filtering_options = StringField(
+            label="Filtering expressions",
+            description="Filter VCF file using arbitraty expressions."
+            "Examples of filtering expressions: '(ANN[*].GENE = 'PSD3')' "
+            "or '( REF = 'A' )' or "
+            "'(countHom() > 3) | (( exists INDEL ) & (QUAL >= 20)) | (QUAL >= 30 )'."
+            "For more information checkout the official documentation of [SnpSift]"
+            "(https://pcingola.github.io/SnpEff/ss_filter/)",
+            required=False,
+        )
+        sets = ListField(
+            DataField(data_type="file"),
+            label="Files with list of genes",
+            description="Use list of genes, if you only want variants reported for "
+            "them. Each file must have one string per line.",
+            hidden="!filtering_options",
+            required=False,
+        )
+        extract_fields = ListField(
+            StringField(),
+            label="Fields to extract",
+            description="Write fields you want to extract from annonated vcf file "
+            "and press Enter after each one. Example of fields: `CHROM POS REF ALT "
+            "'ANN[*].GENE'`. For more information follow this [link]"
+            "(https://pcingola.github.io/SnpEff/ss_extractfields/).",
+            required=False,
+        )
+
+        class Advanced:
+            """Advanced options."""
+
+            one_per_line = BooleanField(
+                label="One effect per line",
+                default=False,
+                description="If there is more than one effect per variant, write them "
+                "to seperate lines.",
+            )
+
+        advanced = GroupField(
+            Advanced, label="Advanced options", hidden="!extract_fields"
+        )
+
+    class Output:
+        """Output fields to process SnpEff."""
+
+        annotation = FileField(
```

Comment:
Since the type of the process is a `"data:variants:vcf:snpeff"`, the output field structure should match all other `"data:variants:vcf:*"` processes (`vcf`, `tbi`, `species`, `build`).